src/context/user-products-context.jsx

```diff
@@ -0,0 +1,46 @@
+import { createContext, useContext, useReducer } from "react";
+
+const UserProductsContext = createContext();
+
+const useUserProducts = () => useContext(UserProductsContext);
+
+const initialUserProducts = { wishlist: [], cart: [] };
+
+const userProductReducer = (state, action) => {
```

Comment:
type and payload can be destructured here 

`( state, { type, payload } )`

Reply:
okay, will do it later during abstraction